source/Pggendtoend.test.cc

```diff
@@ -0,0 +1,131 @@
+          string(" updates = ") + to_string(updates) + 
+          string(" sym_limit = ") + to_string(sym_limit) + 
+          string(" lysis = ") + to_string(lysis) + 
+          string(" horiz_trans = ") + to_string(horiz_trans) + 
+          string(" burst_size = ") + to_string(burst_size) + 
+          string(" burst_time = ") + to_string(burst_time) + 
+          string(" host_repro_res = ") + to_string(host_repro_res) + 
+          string(" sym_lysis_res = ") + to_string(sym_lysis_res) + 
+          string(" sym_horiz_trans_res = ") + to_string(sym_horiz_trans_res) +
+	 string(" start_moi = ") + to_string(start_moi) +
+          string(" grid = ") + to_string(grid) +
+          string("Pgg_donation = ")+to_string(donation)+"}") {
+    
+    config.Write("SymSettings.cfg");
+
+    THEN( "Symublation runs without error" ) {
+      symbulation_main(0, NULL);
+    }
+
+    string type;
+    for(int i = 0; i < 2; i++) {
+      if(i == 0)
+        type = "SymVals";
+      else
+        type = "HostVals";
+
+      string path = "source/Pgg_end_to_end_test_data/"+type+"_"+expected_result_file+".data";
+
+      THEN( "Symbulation's actual "+type+" output found at \"" + type + "_Test.data\" matches the expected output at \"" + path + "\"" ) {
+        ifstream actual, expected;
+        actual.open(type + "_Test.data", ios::in);
+        expected.open(path, ios::in);
+	REQUIRE(actual.is_open());
+        REQUIRE(expected.is_open());
+      
+        //Is this a good length of text to display to the user?
+        const int chunk_length = 1000;
+
+        char actual_buffer[chunk_length];
+        char expected_buffer[chunk_length];
+
+        while (!(actual.eof() && expected.eof())) {
+
+          actual >> actual_buffer;
+          expected >> expected_buffer;
+
+          //This provides a nice error with context when they don't match.
+          // array<char, chunk_length> actual_chunk;
+          // move(begin(actual_buffer), end(actual_buffer), actual_chunk.begin());
+          // array<char, chunk_length> expected_chunk;
+          // move(begin(expected_buffer), end(expected_buffer), expected_chunk.begin());
+          string actual_chunk = string(actual_buffer);
+          string expected_chunk = string(expected_buffer);
+
+
+          REQUIRE(actual_chunk == expected_chunk);
+
+        }
+
+        actual.close();
+        expected.close();
+      }//THEN
+    }//type loop
+  }//GIVEN
+}//void Test(.........)
+
+/*Test(std::string expected_result_file, int seed = 10, int data_int = 100, double mutation_rate = 0.002, 
+    double synergy = 5, double vertical_transmission = 1, double host_int = 0, 
+    double sym_int = 0,  int grid_x = 5, int grid_y = 5, int updates = 1, 
+    int sym_limit = 1, bool lysis = 0, bool horiz_trans = 0, int burst_size = 10, 
+    int burst_time = 10, double host_repro_res = 1000,double sym_lysis_res = 1, 
+	  double sym_horiz_trans_res = 100, int start_moi = 1, bool grid = 0)
+*/
+TEST_CASE( "End To End" ) {
```

Comment:
It doesn't look like data files are actually being added to check against, so I don't think this test will do anything. I also think we are going to try to move away from this kind of test since it is difficult to maintain, so I would prefer you remove it.